### PR TITLE
Add Eu isolation Keyboard support

### PR DESCRIPTION
**Pull request by aadriance**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* This PR takes the open source firmware for EU-isolation and cleans it up for merging into qmk.
* See https://github.com/TuckTuckFloof/qmk_firmware/commit/c077753e114f1476b88176c1db23442238946bf2 for original source
* Firmware flashed to my keyboard to confirm usability

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by aadriance** [comment]
@TuckTuckFloof as an FYI incase you have any input on merging this in.  Just thought it would be worth sharing it to the QMK community.

**Comment by aadriance** [comment]
After reading the PR checklist I realized I could test the configurator layout file.  Did that and got it updated because it initially looked weird. Pardon the extra commit

File: keyboards/eu_isolation/info.json [hunk 1]
```diff
@@ -0,0 +1,171 @@
+{
+    "keyboard_name": "EUISOlation",
+    "url": "",
+    "maintainer": "qmk",
+    "width": 14,
+    "height": 4,
```
Comment (drashna) [inline keyboards/eu_isolation/info.json:6]:
```suggestion
```

File: keyboards/eu_isolation/keymaps/bigspace/config.h [hunk 2]
```diff
@@ -0,0 +1,2 @@
+#define COMBO_COUNT 5
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/bigspace/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

And it needs an include header (`#pragma once`)

File: keyboards/eu_isolation/keymaps/bigspace/keymap.c [hunk 3]
```diff
@@ -0,0 +1,45 @@
+#include "eu_isolation.h"
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/bigspace/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance:
```c
// Copyright 2021 Your Name (@yourgithub)
// SPDX-License-Identifier: GPL-2.0-or-later
```

Or 

https://github.com/qmk/qmk_firmware/blob/6b299fab6e6a3143c061b9ea005253361fcd524b/data/templates/base/keyboard.c#L1-L15

Also, 
```suggestion
#include QMK_KEYBOARD_H
```

File: keyboards/eu_isolation/keymaps/bigspace/keymap.c [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+enum layers {
+    _BASE = 0,
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/bigspace/keymap.c:4]:
```suggestion
    _BASE,
```

File: keyboards/eu_isolation/keymaps/default/config.h [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#define COMBO_COUNT 5
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/default/config.h:1]: Same as above

File: keyboards/eu_isolation/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,45 @@
+#include "eu_isolation.h"
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/default/keymap.c:1]: Same as above

File: keyboards/eu_isolation/keymaps/mit/config.h [hunk 7]
```diff
@@ -0,0 +1,2 @@
+#define COMBO_COUNT 5
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/mit/config.h:1]: Same as above

File: keyboards/eu_isolation/keymaps/mit/keymap.c [hunk 8]
```diff
@@ -0,0 +1,45 @@
+#include "eu_isolation.h"
```
Comment (drashna) [inline keyboards/eu_isolation/keymaps/mit/keymap.c:1]: Same as above.

File: keyboards/eu_isolation/readme.md [hunk 9]
```diff
@@ -0,0 +1,11 @@
...
+![euisolation](https://i.imgur.com/5LGsiJZ.jpg)
+
+EU ISOlation is an ortholinear 40% keyboard designed to use up your spare iso
+enters from you collection of keyboard kits.
+
+* Keyboard Maintainer: Originally by [TuckTuckFloof](https://github.com/TuckTuckFloof)
+ cleaned up and merged to QMK by [aadriance](https://github.com/aadriance)
+* Hardware Supported: EU ISOlation
+* Hardware Availability: Originally a [P3DStore](https://p3dstore.com/) group buy. 
+ Open source  bits available on [TuckTuckFloofs github](https://github.com/TuckTuckFloof/EUISO-pcb)
```
Comment (drashna) [inline keyboards/eu_isolation/readme.md:11]:
```suggestion

## Bootloader

Enter the bootloader in 3 ways:

* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available

## Information

Make example for this keyboard (after setting up your build environment):

    make eu_isolation:default

Flashing example for this keyboard:

    make eu_isolation:default:flash

See the [build environment setup](https://docs.qmk.fm/#/getting_started_build_tools) and the [make instructions](https://docs.qmk.fm/#/getting_started_make_guide) for more information. Brand new to QMK? Start with our [Complete Newbs Guide](https://docs.qmk.fm/#/newbs).
```

File: keyboards/eu_isolation/rules.mk [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+MIDI_ENABLE = no            # MIDI controls
+AUDIO_ENABLE = no           # Audio output on port C6
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+RGBLIGHT_ENABLE = no        # Enable WS2812 RGB underlight.
+API_SYSEX_ENABLE = no
+
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
```
Comment (drashna) [inline keyboards/eu_isolation/rules.mk:25]:
```suggestion
# MCU name
MCU = atmega32u4

# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = yes           # Enable N-Key Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
AUDIO_ENABLE = no           # Audio output
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by aadriance** [comment]
pushed on big update to address @drashna comments (Thanks for review). I recompiled all firmware and reflashed to my eu-isolation to confirm all is well (this comment brought to you by this firmware :)). I additionally re-opened the json layout in the configurator to ensure that didn't break aswell.

File: keyboards/eu_isolation/rules.mk [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = no       # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = yes           # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+AUDIO_ENABLE = no           # Audio output
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
```
Comment (fauxpark) [inline keyboards/eu_isolation/rules.mk:18]:
```suggestion
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
AUDIO_ENABLE = no           # Audio output
```

File: keyboards/eu_isolation/readme.md [hunk 12]
```diff
@@ -0,0 +1,30 @@
+# EU ISOlation
+![euisolation](https://i.imgur.com/5LGsiJZ.jpg)
```
Comment (fauxpark) [inline keyboards/eu_isolation/readme.md:2]:
```suggestion

![euisolation](https://i.imgur.com/5LGsiJZh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Review by fauxpark — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**